pkg/config/sdk.go

```diff
@@ -55,19 +55,18 @@ func (s *SdkInfo) UnmarshalJSON(data []byte) (err error) {
 			return fmt.Errorf("invalid Stack Layer received %s", string(a.Layer))
 		}
 	}
+	if !a.Layer.Validate() {
```

Comment:
The overall PR looks great.
However, I'd like to highlight a potential issue with the placement of the stack layer validity check. 
In the scoring apps, for example, the SDK config for MQTT (located at `cmd/subscriber/res/docker/config-mqtt.json`) does not include a `layer` field. 
Consequently, when the stack layer check is outside the annotators' length condition, it may erroneously flag an empty stack layer as invalid.
I think the need for a `layer` field is linked to the presence of annotators in the config. If annotators are defined, a corresponding layer is expected. Conversely, in the absence of annotators, the `layer` field is unnecessary. 
Therefore, I suggest we ensure that the stack layer check is only applied when annotators are present to avoid any false negatives

Reply:
@michaelehab Good catch. Not sure what happened here and why that line is duplicated since the check is already in the annotators block above. Will remove.

Reply:
I'm seeing odd behavior in the online diff tool where the change I made in relation to this comment is not being accurately reflected. Here is a direct link to my branch reflecting the change made.

https://github.com/tsconn23/alvarium-sdk-go/blob/b7d49e810f431d92918da724b6e0352b0f39778c/pkg/config/sdk.go#L46